sktime/transformations/compose.py

```diff
@@ -663,3 +664,135 @@ def _underscore_join(iterable):
         """Create flattened column names from multiindex tuple."""
         iterable_as_str = [str(x) for x in iterable]
         return "__".join(iterable_as_str)
+
+
+class FitInTransform(BaseTransformer):
+    """Transformer composition to always fit a given transformer on the transform data only.
+
+    Some transformers like Imputer(method="mean") might be more accurate for some use
```

Comment:
```suggestion
    In panel settings, e.g., time series classification, it can be preferable (or, necessary) to fit and transform on the test set, e.g., interpolate within the same series that interpolation parameters are being fitted on.
    `FitInTransform` can be used to wrap any transformer to ensure that `fit` and `transform` happen always on the same series, by delaying the `fit` to the `transform` batch.
    Warning: use of `FitInTransform` will typically not be useful, or can constitute a mistake (data leakage) when naively used in a forecasting setting.
```